### PR TITLE
Notify issues.labeled required_labels

### PR DESCRIPTION
**Pull request by otoyo**
Fixes https://github.com/integrations/slack/issues/965 
Fixes https://github.com/integrations/slack/issues/963

This PR allows to notify `issues.labeled` events when we use labels filter.
Here is the behavior.

### Before

1. We subscribes `issues` events by using labels filter like `/github subscribe org/repo +label:enhancement`
2. We add the required_label `enhancement` to existing issue, but the integration does not notify this.

### After

1. Same as the above
2. We add the required_label `enhancement` to existing issue, then the integration notifies this.

Note,
if we do not use the labels filter, the integration behaves same as before.

### CODE REVIEW

**Comment by aramando** [comment]
@otoyo Just to confirm, will this apply only to issues, or pull requests too?

**Comment by otoyo** [comment]
Hello @aramando , this diffs make only issues notify.

**Review by prashkan — APPROVED** [review]

**Comment by 3nvi** [comment]
Is there a chance to merge this since it's approved more than a month ago?

**Comment by andrewetchen** [comment]
> Is there a chance to merge this since it's approved more than a month ago?

Hi @prashkan, do you know when this PR will be merged? We've got a bunch of folks who had been patiently waiting for this, thanks.

**Comment by valerybugakov** [comment]
Waiting for this feature to be merged too. Anyway we can help to move it forward?

**Comment by saso-doordash** [comment]
Any updates or next steps on this PR?

**Comment by valerybugakov** [comment]
@otoyo, can we help somehow to move this forward?

**Comment by otoyo** [comment]
This patch stands on the legacy Slack app.
And the legacy app depends on the "workspace tokens" and it will retire in August 2021.
https://api.slack.com/legacy/workspace-apps/working-with-workspace-tokens

I tried to run this patch locally but couldn't for the various reasons, Slack API changes, etc.
(1 year has passed since I opened this PR...

In addition, maintainers of this integration are not welcome any patches now.
https://github.com/integrations/slack#contributing

So please wait for the changes by maintainers.

**Comment by valerybugakov** [comment]
Thanks for the response, @otoyo.

**Pull request closed without merging**
